lib/puppet/configurer.rb

```diff
@@ -383,36 +382,20 @@ def run_internal(options)
   private :run_internal
 
   def find_functional_server()
-    configured_environment = Puppet[:environment] if Puppet.settings.set_by_config?(:environment)
-
-    node = nil
-    selected_server = Puppet.settings[:server_list].find do |server|
-      # Puppet.override doesn't return the result of its block, so we
-      # need to handle this manually
-      found = false
-      server[1] ||= Puppet[:masterport]
-      Puppet.override(:server => server[0], :serverport => server[1]) do
-        begin
-          node = Puppet::Node.indirection.find(Puppet[:node_name_value],
-              :environment => Puppet::Node::Environment.remote(@environment),
-              :configured_environment => configured_environment,
-              :ignore_cache => true,
-              :transaction_uuid => @transaction_uuid,
-              :fail_on_404 => false)
-          found = true
-        rescue  => detail
-          #TRANSLATORS 'server_list' is the name of a setting and should not be translated
-          Puppet.debug _("Unable to connect to server from server_list setting: %{detail}") % {detail: detail}
-        end
+    Puppet.settings[:server_list].each do |server|
+      host = server[0]
+      port = server[1] || Puppet[:masterport]
+      begin
+        http = Puppet::Network::HttpPool.http_ssl_instance(host, port)
+        response = http.get('/status/v1/simple')
```

Comment:
@Magisus @justinstoller according to https://puppet.com/docs/puppetserver/5.0/status-api/v1/simple.html the simple status endpoint is available in puppetserver 5.0, but wanted to see if there are any gotchas we need to be aware of when backporting this fix to puppet 5.5.x?

Reply:
The status service is available for users to allow access to from 5.0 onwards. However the endpoint is behind a default deny all auth rule until 5.3.6. (current supported 5.x is 5.3.8, fwiw)

Reply:
hmm maybe we should consider 404 to be success (below)?

Reply:
I suppose that would be the most conservative change. Folks shouldn't run `puppet-agent` versions that are newer than their masters, but we also haven't historically made these sorts of behavior changes in a .z release.

Reply:
Although, it may be a 403 error instead of a 404.

Reply:
Another option is to provide release notes about allowing requests to the simple server status endpoint?

Reply:
@Sharpie Josh and I talked about this yesterday and are considering just merging this as is without a 403, and just covering it release notes, since this endpoint has had support for a few y-versions prior to where we're merging it. Do you think that's reasonable?

Reply:
I think that's fine. We just need to make it very clear in the notes that these 5.5.z agents should be used with a server that is also 5.5.z vintage.

Reply:
If the agent is configured to use `server_list`, then with this PR, the agent will require server 5.3.6 or later. However, if using server 5.0-5.3.5, and the simple server status fails, will the agent fallback to `server` or will it consider all servers down? The latter would be bad I'm thinking, and maybe we should special case 403 based on the `X-PUPPETSERVER-VERSION` header (or whatever it is)?

Reply:
Ugh, I forgot that Puppet Server versioning is different than Puppet Agent versioning. 5.0 -- 5.3.6 is actually the overwhelming majority of Puppet Server 5.x releases. Only 5.3.7 and 5.3.8 would be exempt from the issue.

So, we should special case the 403 --- otherwise we risk breaking things on upgrade in a way that shouldn't happen with a .z release.